[NNC] add support for masked_fill

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48975 [NNC] add fusion support for nan_to_num
* #48976 [NNC] Add fusion support of aten::to
* **#48974 [NNC] add support for masked_fill**
* #48973 [NNC] Add Support For is_nan

Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

Commit:
add masked_fill

[ghstack-poisoned]

Commit:
Update on "add masked_fill"




[ghstack-poisoned]

Commit:
Update on "[NNC] add support for masked_fill"


Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

[ghstack-poisoned]

bertmaher commented on torch/csrc/jit/passes/tensorexpr_fuser.cpp:
```diff
@@ -116,6 +116,8 @@ bool isSupported(Node* node) {
       "aten::round(Tensor self) -> Tensor",
       "aten::trunc(Tensor self) -> Tensor",
       "aten::threshold(Tensor self, Scalar threshold, Scalar value) -> Tensor",
+      "aten::masked_fill.Scalar(Tensor self, Tensor mask, Scalar value) -> Tensor",
```
Did you test the scalar flavor?

bertmaher commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -894,6 +896,21 @@ Tensor* TensorExprKernel::computeValue(const torch::jit::Value* v) {
     } break;
 
+    case aten::masked_fill: {
+      return computeThreeOperand(
```
Just checking: does `masked_fill` obey standard broadcasting rules as implied by `computeThreeOperand`?  (Meaning -- I think -- that all of `input`, `mask`, and `value` are broadcasted to the same shape?

eellison replied:
They are all broadcasted to `input`. Since we have guarded based on the sizes,  we know that we have a successful execution and that `mask` and `value` are broadcastable to `input`. I suspect there are a number of ops for which the default `broadcastInputs` logic would need to be refined if we didnt know that we have a valid set of input shapes.

bertmaher reviewed (approved):
lgtm, just a couple questions

Commit:
Update on "[NNC] add support for masked_fill"


Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

[ghstack-poisoned]

Commit:
Update on "[NNC] add support for masked_fill"


Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

[ghstack-poisoned]

Commit:
Update on "[NNC] add support for masked_fill"


Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

[ghstack-poisoned]

Commit:
Update on "[NNC] add support for masked_fill"


Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

[ghstack-poisoned]

Commit:
Update on "[NNC] add support for masked_fill"


Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

[ghstack-poisoned]

Commit:
Update on "[NNC] add support for masked_fill"


Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

[ghstack-poisoned]

Commit:
Update on "[NNC] add support for masked_fill"


Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

[ghstack-poisoned]

Commit:
Update on "[NNC] add support for masked_fill"


Differential Revision: [D25413165](https://our.internmc.facebook.com/intern/diff/D25413165)

[ghstack-poisoned]
